code/modules/mob/living/status_procs.dm

```diff
@@ -302,7 +302,7 @@
 /mob/living/proc/Sleeping(amount, updating = TRUE, ignore_canstun = FALSE) //Can't go below remaining duration
 	if(SEND_SIGNAL(src, COMSIG_LIVING_STATUS_SLEEP, amount, updating, ignore_canstun) & COMPONENT_NO_STUN)
 		return
-	if((!HAS_TRAIT(src, TRAIT_SLEEPIMMUNE)) || ignore_canstun)
+	if(ignore_canstun)
```

Comment:
Check what might cause ignore_stun to be passed in as true, if it's used at all pass it in to IsSleeping() and if it's true, ignore the trait check.

Reply:
Yep, these checks need to go, all of them. The proc's parameter as well.